Assets/Scripts/Controllers/HistoryController.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Controllers;
+using Services.History;
+
+namespace Controllers
+{
+    public class HistoryController: MonoBehaviour
+    {
+        private Stack<ICommand> _Undocommands = new Stack<ICommand>();
```

Comment:
Separate unity specific stuff (Controller based) from the business logic (Services).

First a little backstory about code modularization & architecture: 
A user action history is a service that records actions and allows to undo & redo them (and so those would be its public methods). A history controller is a client that uses this services functionality by implementing user-action interface to fit clients needs (e.g. undoing user node selections - note unity context).
So in this case a `HistoryController` tasks are to:
- Subscribe to `SelectedNode` changes
- Create a user-action implementation
- Register it in service

On top of that it needs to inform action implementation how it should handle it's execution - so it binds its static delegate/Action to the appropriate function in `NodeController` (read about it further down). This way the service layer stays independent & clueless about what its doing.